src/Lecture3.hs

```diff
@@ -138,6 +162,8 @@ data List1 a = List1 a [a]
 
 -- | This should be list append.
 instance Semigroup (List1 a) where
+  (<>) (List1 x []) (List1 y ys) = List1 x (y : ys)
```

Comment:
You don't need this extra case because `[] ++ b` will be evaluated to `b` and this is already handled by `++`

Reply:
I see, thanks. It is more clear now.